bids2table/entities.py

```diff
@@ -248,7 +213,7 @@ def to_path(
 
     def with_update(
         self, entitities: Optional[Dict[str, Any]] = None, **kwargs
-    ) -> "BIDSEntities":
+    ) -> "_BIDSEntitiesBase":
```

Comment:
You can use `Self` to indicate that the output has the same type as input.

```suggestion
    ) -> Self:
```